dts/bindings/mtd/nordic,qspi-nor.yaml

```diff
@@ -27,6 +27,23 @@ properties:
   quad-enable-requirements:
     default: "S1B6"
 
+  # Some chips don't put the Quad Enable bit in the same register while
+  # otherwise being completely compatible, give option for override
+  quad-enable-bit-mask:
```

Comment:
A property for specifying this bit already exists, see [quad-enable-requirements](https://github.com/zephyrproject-rtos/zephyr/blob/83939c267c1a4ffe2644a913da79b5409e114f81/dts/bindings/mtd/jedec%2Cjesd216.yaml#L41-L58). The nrf_qspi_nor driver currently supports only the `S1B6` value that is [set as default](https://github.com/zephyrproject-rtos/zephyr/blob/83939c267c1a4ffe2644a913da79b5409e114f81/dts/bindings/mtd/nordic%2Cqspi-nor.yaml#L26-L28) for `"nordic,qspi-nor"` nodes.
#42049 adds support for the `S2B1v4` value. It seems that it should work for GD25Q16C, too.